pkg/dnservice/cfg.go

```diff
@@ -57,6 +59,11 @@ type Config struct {
 		ClientConfig logservice.HAKeeperClientConfig
 	}
 
+	// Frontend parameters for the frontend, assign by mo-service's Config
+	Frontend config.FrontendParameters
+	// ETLFSFactory create FS for trace/metric to do CSV output
+	ETLFSFactory fileservice.FileServiceFactory
```

Comment:
This is not needed.
If the provided path has a service name, you should get an ETLFileService with fileservice.Get
If not, the path may be a local file system path, call NewLocalETLFS to get one
Like here: https://github.com/matrixorigin/matrixone/blob/15738771b225a3a17940b8a4f58a13bb0765da10/pkg/sql/colexec/external/external.go#L139

Reply:
seems `ReadFile` have problem if the external table belong to user's S3. @jianwan0214

Reply:
Users can define their S3 services in the config file, like
```
[[fileservice]]
name = "mys3"
backend = "S3"
shared-config-profile = "mys3-profile"
```
Then use the 'mys3' service name to refer to files in that S3 instance
```create external table ... infile {"filepath"="mys3:a/b/c"} ```